react-components/src/components/GiftCardOptions/useGiftCardOptions.js

```diff
@@ -25,43 +25,41 @@ const useGiftCardOptions = props => {
     const [giftCardState, setGiftCardState] = useState(null);
     const giftCardProductQuery = useAwaitQuery(GIFT_CARD_PRODUCT_QUERY);
 
-    useEffect(() => {
-        const fetchGiftCardOptions = async () => {
-            if (!sku) {
-                console.error('SKU is not present in the dataset of mountpoint element');
-                return;
-            }
-            const { data, error } = await giftCardProductQuery({ variables: { sku }, fetchPolicy: 'network-only' });
-
-            if (error) {
-                throw new Error(error);
-            }
+    const fetchGiftCardOptions = useCallback(async () => {
```

Comment:
Why is this change necessary ?

Reply:
This returns a function that is updated when the `sku` changes. And again when the function changes `useEffect` is called again to fetch the gift card options. Before the options were fetched only once. 

I used `useCallback` to create a memorised value as it is unlikely that the `sku` actually changes.

Reply:
why not use just the `useEffect` with a dependency on `sku` ?

Reply:
done